docs/source/traitsui_user_manual/testing/discussions/event_loop_and_exceptions.rst

```diff
@@ -0,0 +1,31 @@
+GUI event processing and exception handling
+===========================================
+
+The testing API is designed such that GUI event processing is requested
+automatically wherever there may be pending GUI events. While events are being
+processed, the global exception handling is overridden temporarily in order to
+capture any unhandled exceptions. A test error is resulted if any exceptions
+are caught.
```

Comment:
Maybe mention that even if a test error occurs, the event loop is still cleaned up prior to other tests, which prevents test interactions.